4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+#' for each time period and each GCM, if predictions for multiple GCMs are being 
+#' read in.
+#' @param run_gcm_list A list of GCM names that corresponds to the `site_meteo_files` 
+#' and `export_files` that are being read in. Depending on whether predictions are being
+#' read in for a single lake-gcm combo, or for all GCMs for a single lake, it may 
+#' be a list of the same GCM name repeated, or a list that includes all 6 GCM names.
+#' @return A munged dataframe of GLM predictions that includes columns for date, month, 
+#' year, DOY, decade, period, gcm, ice, hice (ice thickness), evap (evaporation),
+#' n_layers (number of simulated layers), and temperature predictions at 0.5-meter intervals
+get_site_preds <- function(site_meteo_files, site_export_files, run_gcm_list) {
+  site_preds <- purrr::pmap_df(list(site_meteo_files, site_export_files, run_gcm_list), function(raw_meteo_fl, export_file, gcm) {
+    # Define time period begin and end dates from raw meteo_fl
+    meteo_data <- arrow::read_feather(raw_meteo_fl, col_select = "time")
+    begin <- min(meteo_data$time)
+    end <- max(meteo_data$time)
+    # read in data for that time period and truncate
+    arrow::read_feather(export_file) %>%
+      filter(time >= as.Date(begin) & time <= as.Date(end)) %>%
+      mutate(gcm = gcm)
+  }) %>%
+    mutate(date = as.Date(time), .before=1) %>%
+    select(-time) %>%
+    mutate("month" = month(date),
+           "year"= year(date), 
+           "doy"= yday(date),
+           "decade" = sprintf('%ss', year - (year %% 10)),
+           "period" = sprintf('%s - %s', year - (year %% 20), (year - (year %% 20)) + 20),
+           .after=date) %>%
+    filter(!(year == 2000)) %>% #Predictions for this year are incomplete and fall in a different 20-year period, so exclude
+    select(date, month, year, doy, decade, period, gcm, ice, hice, evap, n_layers, everything()) # Place the temperature prediction columns last
+  
+  return(site_preds)
+}
+
+#' @title Get surface, middle, and bottom GLM predictions for all lakes
+#' @description Function to pull and combine GLM predictions for all lakes,
+#' across all 3 simulation periods and all 6 GCMs
+#' @param run_groups The `p2_glm_uncalibrated_run_groups` grouped version 
+#' of the `p2_glm_uncalibrated_runs` output tibble subset to the site_id, 
+#' gcm, time_period, raw_meteo_fl, export_fl, and export_fl_hash columns 
+#' and grouped by site_id and gcm. Then filtered to only groups for which 
+#' glm_success==TRUE for all runs in that group.
+#' @return A munged dataframe of surface, middle, and bottom temperature GLM
+#' predictions for all successfully simulated lakes. The munge table includes 
+#' the columns site_id, gcm, date, month, year DOY, decade, period, gcm, 
+#' ice, hice (ice thickness), evap (evaporation), n_layers (number of simulated 
+#' layers), depth_class (surface, middle, or bottom), depth, and temperature 
+get_surface_middle_bottom_preds <- function(run_groups) {
```

Comment:
I think this should also go to 3_extract, or since just used for prepping data to plot, I would suggest pulling this out into a separate script called `4_visualize/src/plot_data_utility_fxns.R` or something.